### ISSUE TITLE
Jasmine doesn't unregister some spy on IE11

### ISSUE THREAD

**Issue by BenoitZugmeyer** [jasmine/jasmine#1456]
I am using `spyOn(window, 'onerror')` to test some uncaught exceptions. It works well in most browsers, except on IE11 where I get a `Error: <spyOn> : onerror has already been spied upon` in the second test case using this pattern.

## Expected Behavior

Jasmine should remove the spy on `window.onerror` after the first test case.

## Current Behavior

Jasmine isn't removing the spy on `window.onerror` after the first test case.

## Possible Solution

This issue is caused by a bug in IE11 (and probably before, but I guess we don't care anymore):
```js
function foo() {}
window.onerror = foo;
console.log(window.hasOwnProperty("onerror")); // false
delete window.onerror;
console.log(window.onerror === foo); // true
```

Because of this, jasmine doesn't remove the spy in `SpyRegistry`, because it never do `obj[methodName] = originalMethod`.

## Suite that reproduces the behavior (for bugs)

You can reproduce this by running this jsbin in IE11: http://jsbin.com/xumakorezo/1/edit?js,output

```javascript
describe("aa", function() {
  it("bb", function () {
    spyOn(window, "onerror");
  })
  it("cc", function () {
    spyOn(window, "onerror");
  })
});
```

## Your Environment

* Version used: 2.8
* Environment name and version: IE11
* Operating System and version (desktop or mobile): Windows 10

**Comment by sgravrock** [comment]
I was able to reproduce this in IE 10 as well. I assume it affects all the versions that Jasmine supports (8 and later) but I don't have VMs for the older versions readily at hand right now.

`SpyRegistry` currently uninstalls spies by reassigning the original value if there was one and deleting the property otherwise. That's the right strategy in general but it doesn't work for `window.onerror`. We probably need to treat `window.onerror` as a special case and always unspy it by assigning rather than deleting. I'd be happy to review a pull request that does that.

**Comment by sgravrock** [comment]
Fixed on master with b6cc34d.
